Add the minThreshold field

OllieWillette667 opened the pull request:
I added the minthreshold field stuff, it was pretty easy, but I am not sure it works, because I have the ghostError

OllieWillette667 committed:
Add the minThreshold field

I added the minthreshold field stuff, it was pretty easy, but I am not sure it works, because I have the ghostError

OllieWillette667 committed:
Change from minThreshold to threshold

If you are wondering why this change needed to be made, ask @jpwalbran

OllieWillette667 committed:
changed threshold_KEY to THRESHOLD_KEY

very cool. When the original code was written, I had minThreshold for most things, but THRESHOLD_KEY as it is now. Then John decided that it should be 'threshold' instead of 'minThreshold'. So I used the powers that be (ctrl+h) to make the change. But I made a mistake. I changed things to 'Threshold', when it needed to be 'threshold'. So I had to ctrl+h again (Threshold => threshold), and I forgot to be case sensitive, so THRESHOLD_KEY got converted into threshold_KEY. Now I have changed things back, and checkstyle will hopefully pass.

jpwalbran commented on server/src/main/java/umm3601/product/ProductController.java:
```diff
@@ -33,7 +33,7 @@
   private static final String NAME_KEY = "name";
   private static final String COMMENT_KEY = "comment";
   private static final String AMOUNT_KEY = "amount";
```
We're not selling these fields as features, so we don't need them in our code.

jpwalbran commented on server/src/main/java/umm3601/product/ProductController.java:
```diff
@@ -103,8 +103,8 @@ private Bson constructFilter(Context ctx) {
     if (ctx.queryParamMap().containsKey(COMMENT_KEY)) {
       filters.add(regex(COMMENT_KEY,  Pattern.quote(ctx.queryParam(COMMENT_KEY)), "i"));
     }
```
Same as the above.

jpwalbran commented on server/src/main/java/umm3601/product/ProductController.java:
```diff
@@ -103,8 +103,8 @@ private Bson constructFilter(Context ctx) {
+    if (ctx.queryParamMap().containsKey(THRESHOLD_KEY)) {
+      filters.add(regex(THRESHOLD_KEY,  Pattern.quote(ctx.queryParam(THRESHOLD_KEY)), "i"));
     }
     if (ctx.queryParamMap().containsKey(AMOUNT_KEY)) {
```
These two lines should be removed due to the same reasons above.

jpwalbran reviewed (changes requested):
Mostly good, but there are a couple of changes that are going to be necessary.

Shua59 committed:
Removed the form fields for product inconsistent with stories promised

Removed fields comment and amount and altered threshold to be a integer server-side

jpwalbran reviewed (approved):
Looks good.
